lahja/endpoint.py

```diff
@@ -129,6 +174,65 @@ def connect_no_wait(self, loop: Optional[asyncio.AbstractEventLoop] = None) -> N
 
             self._process_item(item, config)
 
+    def _create_external_api(self, ipc_path: pathlib.Path) -> None:
+
+        receiver = EndpointConnector(self)
+
+        ConnectorManager = get_connector_manager()
+
+        ConnectorManager.register('get_connector', callable=lambda: receiver)  # type: ignore
+
+        manager = ConnectorManager(address=str(ipc_path))  # type: ignore
+        server = manager.get_server()   # type: ignore
+        threading.Thread(target=server.serve_forever, daemon=True).start()
+
+    def connect_to_endpoints_blocking(self, *endpoints: ConnectionConfig, timeout: int=30) -> None:
+        """
+        Connect to the given endpoints and block until the connection to every endpoint is
+        established. Raises a ``TimeoutError`` if connections do not become available within
+        ``timeout`` seconds (default 30 seconds).
+        """
+        for endpoint in endpoints:
+            wait_for_path_blocking(endpoint.path, timeout)
+            self._connect_to_endpoint(endpoint)
+
+    async def connect_to_endpoints(self, *endpoints: ConnectionConfig) -> None:
+        """
+        Asynchronously connect to the given endpoints and await until all connections are
+        established.
+        """
+
+        await asyncio.gather(
+            *(self._await_connect_to_endpoint(endpoint) for endpoint in endpoints),
+            loop=self._loop
+        )
+
+    def connect_to_endpoints_nowait(self, *endpoints: ConnectionConfig) -> None:
+        """
+        Connect to the given endpoints as soon as they become available but do not block.
+        """
+        for endpoint in endpoints:
+            asyncio.ensure_future(self._await_connect_to_endpoint(endpoint))
+
+    async def _await_connect_to_endpoint(self, endpoint: ConnectionConfig) -> None:
+        await wait_for_path(endpoint.path)
+        self._connect_to_endpoint(endpoint)
+
+    def _connect_to_endpoint(self, endpoint: ConnectionConfig) -> None:
+        ConnectorManager = get_connector_manager()
+
+        ConnectorManager.register(  # type: ignore
+            'get_connector',
+            proxytype=ProxyEndpointConnector
+        )
+
+        manager = ConnectorManager(address=str(endpoint.path))  # type: ignore
+        manager.connect()
+        self._connected_endpoints[endpoint.name] = manager.get_connector()  # type: ignore
+
+    def is_connected_to(self, endpoint_name: str) -> bool:
```

Comment:
This (and `_connected_endpoints` in general) feels like a minor footgun. Because the caller has control of (name, path) there are no guarantees that names are unique. Maybe `_connected_endpoints` should use `(name, path)` as the key? Or maybe the `connect_to_endpoints_*` methods should check that the name doesn't already exist and throw an error if it does.

Reply:
>Or maybe the `connect_to_endpoints_*` methods should check that the name doesn't already exist and throw an error if it does.

I added that. Because, The currently Lahja always works under the assumption that names are unique (and ensuring that got lost as I dropped `EventBus`). E.g. you can use `BroadcastConfig(filter_endpoint="foo")` to only propagate something to the `foo` endpoint.